packages/discount-code-importer/src/main.js

```diff
@@ -1,3 +1,221 @@
+            return Promise.resolve()
+          })
+          .catch((error) => {
+            if (this.continueOnProblems) {
+              this._summary.updateErrorCount += 1
+              this._summary.errors.push(error)
+              // eslint-disable-next-line max-len
+              const msg = 'Update error occured but ignored. See summary for details'
+              this.logger.warn(msg)
+              return Promise.resolve()
+            }
+            this._summary.updateErrorCount += 1
+            this._summary.errors.push(error)
+            return Promise.reject(error)
+          })
+      return this._create(newCode)
+        .then(() => {
+          this._summary.created += 1
+          return Promise.resolve()
+        })
+        .catch((error) => {
+          if (this.continueOnProblems) {
+            this._summary.createErrorCount += 1
+            this._summary.errors.push(error)
+            // eslint-disable-next-line max-len
+            const msg = 'Create error occured but ignored. See summary for details'
+            this.logger.warn(msg)
+            return Promise.resolve()
+          }
+          this._summary.createErrorCount += 1
+          this._summary.errors.push(error)
+          return Promise.reject(error)
+        })
+    })
+  }
+
+  _update (newCode: CodeData, existingCode: CodeData) {
+    const service = createRequestBuilder({
+      projectKey: this.apiConfig.projectKey,
+    })
+    const actions = this.syncDiscountCodes.buildActions(newCode, existingCode)
+    // don't call API if there's no update action
+    if (!actions.length)
+      return Promise.resolve({ statusCode: 304 })
+    const uri = service.discountCodes.byId(existingCode.id).build()
+    const req = {
+      uri,
+      method: 'POST',
+      body: { version: existingCode.version, actions },
+    }
+    return this.client.execute(req)
+  }
+
+  _create (code: CodeData) {
+    const service = createRequestBuilder({
+      projectKey: this.apiConfig.projectKey,
+    })
+    const uri = service.discountCodes.build()
+    const req = {
+      uri,
+      method: 'POST',
+      body: code,
+    }
+    return this.client.execute(req)
+  }
+
+  _resetSummary () {
+    this._summary = {
+      created: 0,
+      updated: 0,
+      unchanged: 0,
+      createErrorCount: 0,
+      updateErrorCount: 0,
+      errors: [],
+    }
+    return this._summary
+  }
+
+  summaryReport () {
+    const {
+      created,
+      updated,
+      unchanged,
+      createErrorCount,
+      updateErrorCount,
+    } = this._summary
+    let message = ''
+    if (created === 0 && updated === 0 &&
+      createErrorCount === 0 && updateErrorCount === 0)
+      message = 'Summary: nothing to do, everything is fine'
+    else
+      // eslint-disable-next-line max-len
+      message = `Summary: there were ${created + updated} successfully imported discount codes (${created} were new, ${updated} were updates and ${unchanged} were unchanged).`
```

Comment:
I was confused because of those two different forms in the same sentence "were updates" and "were unchanged".